x-pack/legacy/plugins/lens/public/app_plugin/plugin.tsx

```diff
@@ -93,9 +104,54 @@ export class AppPlugin {
             http: core.http,
           })
         );
+        const updateUrlTime = (urlVars: Record<string, string>): void => {
+          const decoded: RisonObject = rison.decode(urlVars._g) as RisonObject;
+          if (!decoded) {
+            return;
+          }
+          // @ts-ignore
+          decoded.time = data.query.timefilter.timefilter.getTime();
+          urlVars._g = rison.encode((decoded as unknown) as RisonObject);
+        };
+        const redirectTo = (
+          routeProps: RouteComponentProps<{ id?: string }>,
+          addToDashboardMode: boolean,
+          id?: string
+        ) => {
+          if (!id) {
+            routeProps.history.push('/lens');
+          } else if (!addToDashboardMode) {
+            routeProps.history.push(`/lens/edit/${id}`);
+          } else if (addToDashboardMode && id) {
+            routeProps.history.push(`/lens/edit/${id}`);
+            const url = context.core.chrome.navLinks.get('kibana:dashboard');
+            if (!url) {
+              return;
+            }
+            const lastDashboardAbsoluteUrl = url.url;
```

Comment:
This also seems like a bunch of detail that belongs to the dashboard module, not to Lens. Maybe something like this:

```js
  const url = dashboardUrl.addEmbeddable(url.url, id, 'lens');
  if (url) {
    window.history.pushState({}, '', url);
  }
```

Reply:
This was the original intention, but then we figured the time range is not properly set. So for the hack with setting timerange to work, this won't be so clean.